### PR TITLE
Document the cluster aggregate module

### PR DESCRIPTION
**Pull request by arbulu89**
Basic documentation of the cluster aggregate.
I have tried to simply specify the core concepts of the aggregate, without going too technical.
At the end, the idea is that anyone coming to this code has an initial knowledge about this aggregate and boundaries.

### CODE REVIEW

File: lib/trento/domain/cluster/cluster.ex [hunk 1]
```diff
@@ -1,5 +1,51 @@
...
+  @moduledoc """
+  The cluster aggregate manages all the domain logic related to
+  deployed HA Clusters (Pacemaker, Corosync, etc).
+  The HA cluster is used to handle the high availability scenarios on the installed
+  SAP infrastructure. That's why this domain is tailored to work on clusters managing
+  SAP workloads. Any other type of resource is not handled properly.
+
+  Each deployed cluster is registered as a new aggregate entry, meaning that all the hosts belonging
+  to the same cluster are part of the same stream. Among all the hosts, only the cluster discovery messages
+  coming from the **designated controller** update the aggregate. In any case, all the hosts are listed in
```
Comment (nelsonkopliku) [inline lib/trento/domain/cluster/cluster.ex:11]: > Among all the hosts, only the cluster discovery messages coming from the **designated controller** update the aggregate

Not really...

We take into account discoveries from non designated controllers to add them to the cluster, if it wasn't already registerd.
We do take into account those only after the designated controller was registered.

```elixir
# If no DC node was received yet, no cluster was registered.
def execute(%Cluster{cluster_id: nil}, %RegisterClusterHost{designated_controller: false}),
  do: {:error, :cluster_not_found}

def execute(
      %Cluster{} = cluster,
      %RegisterClusterHost{
        host_id: host_id,
        designated_controller: false
      }
    ) do
  maybe_emit_host_added_to_cluster_event(cluster, host_id)
end
```

**Review by nelsonkopliku — COMMENTED** [review]
Nice! I noticed a little inconsistency on what happens with the designated controller.

File: lib/trento/domain/cluster/cluster.ex [hunk 2]
```diff
@@ -1,5 +1,51 @@
 defmodule Trento.Domain.Cluster do
-  @moduledoc false
+  @moduledoc """
+  The cluster aggregate manages all the domain logic related to
+  deployed HA Clusters (Pacemaker, Corosync, etc).
+  The HA cluster is used to handle the high availability scenarios on the installed
+  SAP infrastructure. That's why this domain is tailored to work on clusters managing
+  SAP workloads. Any other type of resource is not handled properly.
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply lib/trento/domain/cluster/cluster.ex:7]: Well, in reality they are handled. We have `drbd` clusters for instance. I simply meant that many of the functionalities won't be applied.
I will remove this sentence entirely.

File: lib/trento/domain/cluster/cluster.ex (hunk 1 above)
Comment (arbulu89) [reply lib/trento/domain/cluster/cluster.ex:11]: Ok, you are right. I meant that all the values "except" the host list are only updated in those conditions. That's why I added the next sentence. But I will make it more clear

File: lib/trento/domain/cluster/cluster.ex (hunk 2 above)
[the remark this answers is not in the document]
Comment (arbulu89) [reply lib/trento/domain/cluster/cluster.ex]: I have removed the sentence at the end. Even though other cluster types could be handled, giving this information is not valuable, as they we won't have this scenario

File: lib/trento/domain/cluster/cluster.ex (hunk 1 above)
Comment (arbulu89) [reply lib/trento/domain/cluster/cluster.ex]: Rephrased

**Review by nelsonkopliku — APPROVED** [review]
DOCH! :scroll:

File: lib/trento/domain/cluster/cluster.ex [hunk 3]
```diff
@@ -1,5 +1,52 @@
...
+  Each cluster type has a different way of evaluating the health.
+
+  ### Checks health
+
+  The checks health is obtained from the [Checks Engine executions](https://github.com/trento-project/wanda/).
+  Every time a checks execution is started, the selected checks for this cluster are executed, and based on the result
+  the health value is updated. The checks are started from a user request or periodically following the
+  project scheduler configuration.
+
+  This domain only knows about the health, the details about the execution are stored in the
```
Comment (fabriziosestito) [inline lib/trento/domain/cluster/cluster.ex:47]: just a little change here: "This bounded context" is more accurate.

**Review by fabriziosestito — CHANGES_REQUESTED** [review]
Great work and thanks for this!

Just a mini nitpick on DDD terminology and LGTM

File: lib/trento/domain/cluster/cluster.ex (hunk 3 above)
Comment (nelsonkopliku) [reply lib/trento/domain/cluster/cluster.ex:47]: Question: who is **This bounded context**?

If we refer to **web** or **wanda** I'd be fine, but if we refer to the Cluster Aggregate, I wouldn't agree.

Then what about other references to `domain` in this Aggregate's documentation?

**Review by fabriziosestito — APPROVED** [review]

**Pull request merged**